### PR TITLE
Fix yesterday

### PR DESCRIPTION
**Pull request by AlexeiGitH**
A fix for issue #640
Decided to fix this one since I wrote the extension method.
I changed the method itself as suggested, as well as the test case.
 
## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 4.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [ ] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -45,6 +45,7 @@ The changelog for **SwifterSwift**. Also see the [releases](https://github.com/S
   - `random(from:upTo:)` in favor of `random(in:)` and `random(in:using:)`. [#576](https://github.com/SwifterSwift/SwifterSwift/pull/576) by [guykogus](https://github.com/guykogus)
   - `timeZone` should never have been added because `Date`s are timezone-agnostic. This came to my attention during unit testing over daylight savings changes. [#594](https://github.com/SwifterSwift/SwifterSwift/pull/594) by [guykogus](https://github.com/guykogus)
 ### Removed
+  - removed `firstIndex(where: )`, `firstIndex(of:)`, `lastIndex(where: )`, `lastIndex(of:)` which are no longer needed. [#637](https://github.com/SwifterSwift/SwifterSwift/pull/637) by [marcocapano](https://github.com/marcocapano)
```
Comment (guykogus) [inline CHANGELOG.md]: Why is this here?

File: Sources/Extensions/Foundation/DateExtensions.swift [hunk 2]
```diff
@@ -455,7 +455,9 @@ public extension Date {
     ///     let yesterday = date.yesterday // "Oct 2, 2018, 10:57:11"
     ///
     public var yesterday: Date {
-        return addingTimeInterval(-86400.0)
+        let cal = Calendar.current
+        let yesterday = cal.date(byAdding: .day, value: -1, to: self)
+        return yesterday != nil ? yesterday! : Date()
```
Comment (guykogus) [inline Sources/Extensions/Foundation/DateExtensions.swift]:
```suggestion
        return Calendar.current.date(byAdding: .day, value: -1, to: self) ?? Date()
```

File: CHANGELOG.md (hunk 1 above)
Comment (AlexeiGitH) [reply CHANGELOG.md]: Hm, pulled this from upstream master since my forked one was different.
Not sure why it is counting it as a new file, I did not make this change.
Will remove manually i guess.

**Comment by davedelong** [comment]
This same fix must be applied to `.tomorrow` as well

File: CHANGELOG.md [hunk 3]
```diff
@@ -44,7 +44,6 @@ The changelog for **SwifterSwift**. Also see the [releases](https://github.com/S
 - **Date**
   - `random(from:upTo:)` in favor of `random(in:)` and `random(in:using:)`. [#576](https://github.com/SwifterSwift/SwifterSwift/pull/576) by [guykogus](https://github.com/guykogus)
   - `timeZone` should never have been added because `Date`s are timezone-agnostic. This came to my attention during unit testing over daylight savings changes. [#594](https://github.com/SwifterSwift/SwifterSwift/pull/594) by [guykogus](https://github.com/guykogus)
-### Removed
```
Comment (guykogus) [inline CHANGELOG.md]: Don't remove this line

Comment (AlexeiGitH) [reply CHANGELOG.md]: Fixed

**Comment by LucianoPAlmeida** [comment]
Hey @AlexeiGitH can you add the changelog entry for those fixes?
